### PR TITLE
Removed AndroidSpeechPlayer.UtteranceProgressListener and replaced wi…

### LINKED ISSUE #1003: OnUtteranceProgressListener in AndroidSpeechPlayer not supported by API 14
Looks like the previous listener only fired when utterance was completed. https://developer.android.com/reference/android/speech/tts/TextToSpeech.OnUtteranceCompletedListener

### PR DESCRIPTION
**Pull request by devotaaabel**
…th onUtteranceCompletedListener. Fixes #1003

### CODE REVIEW

**Comment by devotaaabel** [comment]
@Guardiola31337 Can we just not, and revert the last commit? I think we're over-engineering if we're doing all that for future proofing

**Comment by Guardiola31337** [comment]
> I think we're over-engineering if we're doing all that for future proofing

Well, IMO the main issue here is that we shouldn't rely on deprecated APIs ¯\\\_(ツ)\_/¯

**Comment by Guardiola31337** [comment]
What do you think @danesfeder? ☝️

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java [hunk 1]
```diff
@@ -46,6 +49,11 @@ public void onInit(int status) {
   @Override
   public void play(String instruction) {
     if (!isMuted && !TextUtils.isEmpty(instruction)) {
+
+      if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java]: Can we extract this into a private method with naming that explains we are firing the listener only for `ICE_CREAM_SANDWICH_MR1`?

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java [hunk 2]
```diff
@@ -95,27 +103,44 @@ private void muteTts() {
...
-        }
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
+      textToSpeech.setOnUtteranceCompletedListener(new Api14UtteranceListener());
+    } else {
+      textToSpeech.setOnUtteranceProgressListener(new UtteranceListener());
+    }
+  }
+
+
+  class Api14UtteranceListener implements TextToSpeech.OnUtteranceCompletedListener {
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java]: The listeners extracted look 💯 - would we be able to extract them into their own class files to further clean up `AndroidSpeechPlayer`?

**Pull request merged**